### PR TITLE
[12.0][MIG] hr

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration scripts for `hr` module.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# NOTHING TO DO
+
+hr           / hr.job                   / message_last_post (datetime)  : DEL
+hr           / hr.job                   / message_main_attachment_id (many2one): NEW relation: ir.attachment
+hr           / mail.channel             / subscription_department_ids (many2many): NEW relation: hr.department
+# NOTHING TO DO
+
+---XML records in module 'hr'---
+NEW hr.employee: hr.employee_admin (noupdate)
+DEL hr.employee: hr.employee_root (noupdate)
```
Comment (pedrobaeza) [inline addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:32]: We should do something here:

1. We can remove the old `hr.employee_root`, as no longer exists, but probably there will be other records linked to this one.
2. We can rename XML-ID for the new `admin` employee, and update all the proper fields.

Option 2 seems more perfect for the purpose intention.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Also reset template translations. I think all these suggestions deserve my co-authorship on the scripts although I'm not directly typing them (but I can do if that's a problem).

File: addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---Fields in module 'hr'---
+hr           / hr.department            / message_last_post (datetime)  : DEL 
+hr           / hr.department            / message_main_attachment_id (many2one): NEW relation: ir.attachment
+# NOTHING TO DO
+
+hr           / hr.employee              / activity_ids (one2many)       : NEW relation: mail.activity
+hr           / hr.employee              / additional_note (text)        : NEW 
+hr           / hr.employee              / certificate (selection)       : NEW selection_keys: ['bachelor', 'master', 'other']
+hr           / hr.employee              / children (integer)            : previously in module hr_contract
```
Comment (sergio-teruel) [inline addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:9]: @mreficent I think that you have move children and place_of_brith from hr_contract to hr.
https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L1921

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 2 above)
Comment (pedrobaeza) [reply addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:9]: @sergio-teruel This is not needed in this case, as `hr` is loaded before than `hr_contract`, so the XML-ID for `hr` will be created first, and thus no removal will be done.

File: addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 1 above)
Comment (MiquelRForgeFlow) [reply addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:32]: Ok, applied the option 2

**Comment by MiquelRForgeFlow** [comment]
Ready

File: addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 1 above)
Comment (pedrobaeza) [reply addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:32]: But you are not loading in `noupdate_records` the changed fields...

Comment (MiquelRForgeFlow) [reply addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:32]: Yes, in the beginning of the file. Here https://github.com/OCA/OpenUpgrade/pull/1675/files#diff-e962c6851ef25abb2b16c9bb04bf8c9aR3.

Comment (pedrobaeza) [reply addons/hr/migrations/12.0.1.1/openupgrade_analysis_work.txt:32]: OK, sorry. I didn't see it.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**